### ISSUE TITLE
Can you implement "Production Limit" to earn money ;) ?

### ISSUE THREAD

**Issue by richard-e5** [biemond/solaredge.modbus#73]
Hello,

If the electricity is negative I would like to charge my battery. If I could set the "Production limit" to 0 Watt, then my battery can be charged with AC.

Kind regards,
Richard

![IMG_3730](https://github.com/biemond/solaredge.modbus/assets/58510575/642cf827-96da-4c9d-88fb-707f98c656a4)
![IMG_3729](https://github.com/biemond/solaredge.modbus/assets/58510575/553ec977-eaf6-4f64-868c-29ee7743da04)

**Comment by biemond** [comment]
can you try these, it should already be supported

charge from pv, grid.
<img width="462" alt="Screenshot 2024-04-26 at 13 54 41" src="https://github.com/biemond/solaredge.modbus/assets/1085004/d414558b-732f-403c-a337-278d67838356">



<img width="525" alt="Screenshot 2024-04-26 at 13 55 06" src="https://github.com/biemond/solaredge.modbus/assets/1085004/3fce213e-e062-4314-bdff-16b0bb95d7e9">

<img width="627" alt="Screenshot 2024-04-26 at 13 57 00" src="https://github.com/biemond/solaredge.modbus/assets/1085004/7c35af74-aba4-4064-af50-371decea8b1d">


<img width="875" alt="Screenshot 2024-04-26 at 13 57 16" src="https://github.com/biemond/solaredge.modbus/assets/1085004/777cf6f9-a8ba-4b07-91e0-687499f631f0">

**Comment by biemond** [comment]
need to check what production control does but should be almost the same 

```
      if (type == 'limitcontrolmode') {
        // 0 – Disabled
        // 1 – Export Control
        // 2 – Production Control
        // 11 – Minimum Import Control
        if (value == 1) {
          const limitcontrolmodeeRes = await client.writeSingleRegister(0xe000, Number(1));
          console.log('limitcontrolmode', limitcontrolmodeeRes);
          // side = 0
          const limitcontrolsideRes = await client.writeSingleRegister(0xe001, Number(0));
          console.log('limitcontrolside', limitcontrolsideRes);
          // 3000
          const limitcontrolWattRes = await client.writeMultipleRegisters(0xe002, [ 32768, 17723]);
          console.log('limitcontrolwatt', limitcontrolWattRes);
        } else if (value == 11) {
          const limitcontrolmodeeRes = await client.writeSingleRegister(0xe000, Number(2049));
          console.log('limitcontrolmode', limitcontrolmodeeRes);
          // 500
          const limitcontrolWattRes = await client.writeMultipleRegisters(0xe002, [0, 17402]);
          console.log('limitcontrolwatt', limitcontrolWattRes);
        } else {
          const limitcontrolmodeeRes = await client.writeSingleRegister(0xe000, Number(0));
          console.log('limitcontrolmode', limitcontrolmodeeRes);
        }
      }

```

**Comment by richard-e5** [comment]
Thank you for the quick reply.

I tested your setup, but I can not charge the battery with AC.

With Export Limit the battery will charge with PV. With Product Limit the battery wil charge with AC. And if the price is negative I will earn some money ;)

![production-limit](https://github.com/biemond/solaredge.modbus/assets/58510575/10b79d64-c061-4786-8c75-e694ff07cecf)

**Comment by richard-e5** [comment]
yes I can charge from AC, but alway first from PV. I can not choose for AC only. So turn off the SunCells and use only AC.

When I set "Production limit" to 0 Watt (in the SolarEdge install app) then there is no PV, so then the battery can only charge from AC.

![production-limit(2)](https://github.com/biemond/solaredge.modbus/assets/58510575/21fdbd7c-de41-4aca-a6d0-550163c9ebc6)

**Comment by biemond** [comment]
ok, why do you not use Charge from PV+AC?  
can you try active power to 0 , see if that works. 

have to check for production limit.

**Comment by richard-e5** [comment]
I have checked that. Still charge from PV (as picture "With Export Limit" above)

**Comment by richard-e5** [comment]
When I set "Production limit" to 0 Watt, it will charge from AC.

**Comment by biemond** [comment]
nice, will look at it, at least you are unblocked.

by the way, I think price need to be , negative -14 c or not with the transport cost. 

but nice

**Comment by richard-e5** [comment]
Yes I have read that too, but not all think the same. Some say €0 and some say -€0,14. So I have to test that also, when price is negative. But thank you for your information and your thoughs!!

**Comment by richard-e5** [comment]
I asked Zonneplan for the price, when to stop de SolarEdge. The guy said by -€1 ;) 

I tested "Active Power Limit to 0":
1. For an inverter from 2019 it worked.
2. But the new one (2024) with the option "Production Limit" it did not work.

I hope you have an other idear ;) Thanx in advance.

**Comment by biemond** [comment]
I think on the inverter you have to do something to enable active power or not. 

I did not implement production limit,  let me check it out.

**Comment by biemond** [comment]
-1 euro is crazy , you get broke on it.  -14ct should be enough to stop and start importing it.

**Comment by biemond** [comment]
can you try this https://homey.app/a/solaredge.modbus/test/

<img width="394" alt="Screen Shot 2024-05-01 at 20 07 04" src="https://github.com/biemond/solaredge.modbus/assets/1085004/28223ef4-d1ec-4720-8225-ea
<img width="416" alt="Screen Shot 2024-05-01 at 20 07 33" src="https://github.com/biemond/solaredge.modbus/assets/1085004/eb1c22df-ffb5-490d-a516-77350248f86a">
dcb381e360">

**Comment by biemond** [comment]
<img width="394" alt="Screen Shot 2024-05-01 at 20 07 04" src="https://github.com/biemond/solaredge.modbus/assets/1085004/dc0aa5b6-242e-4699-9250-4b93bed70ee8">

**Comment by richard-e5** [comment]
THANX!! Ik ga het opnieuw testen ;)

![afbeelding](https://github.com/biemond/solaredge.modbus/assets/58510575/54910f1e-1fd1-4fc8-8fbd-646c866bccd2)
![afbeelding](https://github.com/biemond/solaredge.modbus/assets/58510575/d1eb9cf3-ad03-424c-a614-a432274cd599)
![afbeelding](https://github.com/biemond/solaredge.modbus/assets/58510575/e9728472-ac4f-45e2-a7f1-0cbdcdd2c7d4)

**Comment by green-us** [comment]
FYI: I got a working power limit on a 2022 HDwave (without battery), indeed by simply setting Active power limit to zero. To set it up I did need to reset the inverter to default settings and enable all modbus and power limiting functions but it was functioning perfectly. However (although nothing on my side changed) this specific option now stopped working. I think since I tested the improved tarif optimisation built in in the MySolaredge app. That works but now the modbus function doesn't anymore. May be related. (topic on tweakers: https://gathering.tweakers.net/forum/list_message/79053502#79053502 .

**Comment by green-us** [comment]
here overview settings screenshots with actually tested and working active power limit manipulation using Homey:
![Screenshot_20240503-154450](https://github.com/biemond/solaredge.modbus/assets/63535613/7afb9448-f170-4d10-ad2b-ae7c9a10a584)
![Screenshot_20240503-154508](https://github.com/biemond/solaredge.modbus/assets/63535613/047c63ae-6cf7-4fa6-accd-9417e2ab85cc)
![Screenshot_20240503-154522](https://github.com/biemond/solaredge.modbus/assets/63535613/31acb482-ad3b-48c2-b6fb-5a371ba098c1)
![Screenshot_20240503-154537](https://github.com/biemond/solaredge.modbus/assets/63535613/a29f0eec-12e1-4338-bb4a-24b841e1840d)
![Screenshot_20240503-154602](https://github.com/biemond/solaredge.modbus/assets/63535613/4b7a090d-92dc-4204-b8c4-5b5d621acde0)
![Screenshot_20240503-154747](https://github.com/biemond/solaredge.modbus/assets/63535613/9f5a1d96-cfe1-4c30-ad55-f3597193949b)
![Screenshot_20240503-154758](https://github.com/biemond/solaredge.modbus/assets/63535613/aaeb5f53-747f-442b-8db6-3381594e79c9)
![Screenshot_20240503-154813](https://github.com/biemond/solaredge.modbus/assets/63535613/e0474b4e-551a-4f9c-8130-0003665fa376)
![Screenshot_20240503-154838](https://github.com/biemond/solaredge.modbus/assets/63535613/09614ca2-41f8-4e59-a368-15a53ad4b6df)
![Screenshot_20240503-154845](https://github.com/biemond/solaredge.modbus/assets/63535613/d8dda1a7-5c3b-408e-8b1a-3f65a8e89bb6)
![Screenshot_20240503-154859](https://github.com/biemond/solaredge.modbus/assets/63535613/fe7347d7-b516-46e8-b70b-923aee424b33)
![Screenshot_20240503-154920](https://github.com/biemond/solaredge.modbus/assets/63535613/643cf251-662d-4296-a22e-cc4a9a5b4485)
![Screenshot_20240503-154945](https://github.com/biemond/solaredge.modbus/assets/63535613/5efa2f42-ee8f-4cfc-a45a-03222d530ac3)
![Screenshot_20240503-154958](https://github.com/biemond/solaredge.modbus/assets/63535613/6f9005d7-1c20-493b-808a-a1b8152401b1)
![Screenshot_20240503-155016](https://github.com/biemond/solaredge.modbus/assets/63535613/d73bac25-5de6-437a-94a9-2fe1372dcce7)
![Screenshot_20240503-155042](https://github.com/biemond/solaredge.modbus/assets/63535613/3973cb4f-294f-4317-aa76-1fb1fa011aa3)
![Screenshot_20240503-155126](https://github.com/biemond/solaredge.modbus/assets/63535613/dd9a35ef-c014-4ed1-9060-e3fb5047d003)

**Comment by biemond** [comment]
Also heard this from others, was working fine until solaredge came with this feature.  I think you are the fourth person talking about this.  I think it is a bug with their new app eature or they want you force you the use the app. 

or when you do it once and go back , something is sticking and auto reverted.

**Comment by green-us** [comment]
Solution (in dutch):

Wifi AP activeren (schakelaar anderhalve tel naar links)
inloggen
naar webgui navigeren (172.16.0.1)
naar 'power control'
vervolgens 'load defaults'
edit: That's enough. 
(de settings onder 'reactive power' mag ook gewoon op default blijven staan (cosphi) in tegenstelling tot mijn eerdere screenshots. Die deed dus niets. Simpelweg het naar default zetten van de power control does the trick.

**Comment by biemond** [comment]
na dit is alles ok voor jou? en active power werkt weer

**Comment by green-us** [comment]
Inderdaad. Lees nu je vraag en probeer het nu nog eens en ja, werkt nog steeds.

**Comment by biemond** [comment]
<img width="748" alt="Screenshot 2024-05-23 at 09 49 56" src="https://github.com/biemond/solaredge.modbus/assets/1085004/07920500-7a49-4e2c-8c96-d774f6c858f2">
